Yeah, I saw. It seems to be because unlike elsewhere, the `np.equal` aval check is called on an array as opposed to a scalar. Calling `array.item()` on each of these turns everything green again.

By the way, locally with `JAX_NUM_GENERATED_CASES=100`:
```
======================================================================
FAIL: testReduce_op=bitwise_and_inshape=int8[3,4,5]_reducedims=(0,)_initval=-1 (__main__.LaxTest)
LaxTest.testReduce_op=bitwise_and_inshape=int8[3,4,5]_reducedims=(0,)_initval=-1
testReduce_op=bitwise_and_inshape=int8[3,4,5]_reducedims=(0,)_initval=-1(op=<function bitwise_and at 0x12f3f6290>, reference_op=<ufunc 'bitwise_and'>, init_val=-1, shape=(3, 4, 5), dtype=<class 'numpy.int8'>, dims=(0,), primitive=reduce_and)
----------------------------------------------------------------------
(Traceback)

AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 4 / 20 (20%)
Max absolute difference: 6
Max relative difference: 1.
 x: array([[0, 1, 0, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0]], dtype=int8)
 y: array([[0, 7, 0, 0, 0],
       [0, 0, 3, 0, 0],
       [2, 0, 0, 0, 0],
       [4, 0, 0, 0, 0]], dtype=int8)

======================================================================
FAIL: testReduce_op=bitwise_and_inshape=uint16[3,4,5]_reducedims=(0,)_initval=-1 (__main__.LaxTest)
LaxTest.testReduce_op=bitwise_and_inshape=uint16[3,4,5]_reducedims=(0,)_initval=-1
testReduce_op=bitwise_and_inshape=uint16[3,4,5]_reducedims=(0,)_initval=-1(op=<function bitwise_and at 0x12f3f6290>, reference_op=<ufunc 'bitwise_and'>, init_val=-1, shape=(3, 4, 5), dtype=<class 'numpy.uint16'>, dims=(0,), primitive=reduce_and)
----------------------------------------------------------------------
Traceback (most recent call last):
[...]
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 3 / 20 (15%)
Max absolute difference: 65534
Max relative difference: 32767.
 x: array([[0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0]], dtype=uint16)
 y: array([[0, 2, 1, 0, 0],
       [0, 0, 0, 0, 2],
       [0, 0, 2, 0, 0],
       [0, 0, 0, 0, 0]], dtype=uint16)

======================================================================
FAIL: testReduce_op=bitwise_and_inshape=uint32[3,4,5]_reducedims=(0,)_initval=-1 (__main__.LaxTest)
LaxTest.testReduce_op=bitwise_and_inshape=uint32[3,4,5]_reducedims=(0,)_initval=-1
testReduce_op=bitwise_and_inshape=uint32[3,4,5]_reducedims=(0,)_initval=-1(op=<function bitwise_and at 0x12f3f6290>, reference_op=<ufunc 'bitwise_and'>, init_val=-1, shape=(3, 4, 5), dtype=<class 'numpy.uint32'>, dims=(0,), primitive=reduce_and)
----------------------------------------------------------------------
Traceback (most recent call last):
[...]
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 2 / 20 (10%)
Max absolute difference: 4294967294
Max relative difference: 2.14748365e+09
 x: array([[0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0]], dtype=uint32)
 y: array([[0, 2, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 1, 0, 0],
       [0, 0, 0, 2, 0]], dtype=uint32)

======================================================================
FAIL: testReduce_op=bitwise_and_inshape=uint8[3,4,5]_reducedims=(0,)_initval=-1 (__main__.LaxTest)
LaxTest.testReduce_op=bitwise_and_inshape=uint8[3,4,5]_reducedims=(0,)_initval=-1
testReduce_op=bitwise_and_inshape=uint8[3,4,5]_reducedims=(0,)_initval=-1(op=<function bitwise_and at 0x12f3f6290>, reference_op=<ufunc 'bitwise_and'>, init_val=-1, shape=(3, 4, 5), dtype=<class 'numpy.uint8'>, dims=(0,), primitive=reduce_and)
----------------------------------------------------------------------
Traceback (most recent call last):
[...]
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 20 (5%)
Max absolute difference: 254
Max relative difference: 127.
 x: array([[0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0]], dtype=uint8)
 y: array([[0, 0, 0, 0, 0],
       [0, 0, 0, 2, 0],
       [0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0]], dtype=uint8)
```

Do you think those are flaky tests, or is there something deeper going on? (There are some other failures, but those are because of disabled x64 mode.)

--- a/jax/_src/lax/lax.py
+++ b/jax/_src/lax/lax.py
@@ -1016,21 +1016,29 @@ def _get_monoid_reducer(monoid_op: Callable,
   aval = core.get_aval(x)
   dtype = _dtype(x)
   if (type(aval) is ConcreteArray) and aval.shape == ():
+    # allow bitwise reductions for boolean and integer types
+    _is_intlike = dtype == np.bool_ or dtypes.issubdtype(dtype, np.integer)
     if monoid_op is add:
       return np.equal(aval.val, 0) and partial(_reduce_sum)
     elif monoid_op is mul:
       return np.equal(aval.val, 1) and _reduce_prod
-    elif monoid_op is bitwise_or and dtype == np.bool_:
-      return np.equal(aval.val, _get_max_identity(dtype)) and _reduce_or
-    elif monoid_op is bitwise_and and dtype == np.bool_:
-      return np.equal(aval.val, _get_min_identity(dtype)) and _reduce_and
-    elif monoid_op is bitwise_xor and dtype == np.bool_:
-      return np.equal(aval.val, _get_max_identity(dtype)) and _reduce_xor
+    elif monoid_op is bitwise_or and _is_intlike:
+      return np.equal(aval.val, _get_bitwise_or_identity(dtype)) and _reduce_or
+    elif monoid_op is bitwise_and and _is_intlike:
+      return np.equal(aval.val, _get_bitwise_and_identity(dtype)) and _reduce_and
+    elif monoid_op is bitwise_xor and _is_intlike:
+      return np.equal(aval.val, _get_bitwise_or_identity(dtype)) and _reduce_xor
     elif monoid_op is max:
       return np.equal(aval.val, _get_max_identity(dtype)) and _reduce_max
     elif monoid_op is min:
       return np.equal(aval.val, _get_min_identity(dtype)) and _reduce_min
   return None
+
+def _get_bitwise_and_identity(dtype: DType) -> Array:
+  return np.array(-1, dtype)
+
+def _get_bitwise_or_identity(dtype: DType) -> Array:
+  return np.array(0, dtype)
 
 def _get_max_identity(dtype: DType) -> Array:
   if dtypes.issubdtype(dtype, np.inexact):
@@ -3761,25 +3769,24 @@ mlir.register_lowering(argmax_p, mlir.cache_lowering(mlir.lower_fun(
 
 
 def _reduce_logical_shape_rule(operand, *, axes):
-  if operand.dtype != np.bool_:
-    msg = "logical reduction requires operand dtype bool, got {}."
-    raise TypeError(msg.format(operand.dtype))
+  if operand.dtype != np.bool_ and not np.issubdtype(operand.dtype, np.integer):
+    raise TypeError(f"logical reduction requires operand dtype bool or int, got {operand.dtype}.")
   return tuple(np.delete(operand.shape, axes))
 
 reduce_or_p = standard_primitive(
-    _reduce_logical_shape_rule, _fixed_dtype(np.bool_), 'reduce_or',
+    _reduce_logical_shape_rule, _input_dtype, 'reduce_or',
     weak_type_rule=_strip_weak_type)
 batching.defreducer(reduce_or_p)
 
 
 reduce_and_p = standard_primitive(
-    _reduce_logical_shape_rule, _fixed_dtype(np.bool_), 'reduce_and',
+    _reduce_logical_shape_rule, _input_dtype, 'reduce_and',
     weak_type_rule=_strip_weak_type)
 batching.defreducer(reduce_and_p)
 
 
 reduce_xor_p = standard_primitive(
-    _reduce_logical_shape_rule, _fixed_dtype(np.bool_), 'reduce_xor',
+    _reduce_logical_shape_rule, _input_dtype, 'reduce_xor',
     weak_type_rule=_strip_weak_type)
 batching.defreducer(reduce_xor_p)
 
@@ -3802,11 +3809,11 @@ mlir.register_lowering(reduce_sum_p, partial(_unary_reduce_lower, mhlo.AddOp,
 mlir.register_lowering(reduce_prod_p, partial(_unary_reduce_lower, mhlo.MulOp,
                                           lambda dtype: np.array(1, dtype)))
 mlir.register_lowering(reduce_or_p, partial(_unary_reduce_lower, mhlo.OrOp,
-                                         lambda dtype: np.array(False, dtype)))
+                                          _get_bitwise_or_identity))
 mlir.register_lowering(reduce_and_p, partial(_unary_reduce_lower, mhlo.AndOp,
-                                          lambda dtype: np.array(True, dtype)))
+                                          _get_bitwise_and_identity))
 mlir.register_lowering(reduce_xor_p, partial(_unary_reduce_lower, mhlo.XorOp,
-                                          lambda dtype: np.array(False, dtype)))
+                                          _get_bitwise_or_identity))
 mlir.register_lowering(reduce_min_p, partial(_unary_reduce_lower, mlir.min_mhlo,
                                          _get_min_identity))
 mlir.register_lowering(reduce_max_p, partial(_unary_reduce_lower, mlir.max_mhlo,

--- a/tests/lax_test.py
+++ b/tests/lax_test.py
@@ -183,6 +183,27 @@ LAX_OPS = [
     op_record("lt", 2, default_dtypes, jtu.rand_small),
 ]
 
+ReducerOpRecord = collections.namedtuple(
+  "ReducerOpRecord", ["op", "reference_op", "init_val", "dtypes", "primitive"]
+)
+
+LAX_REDUCE_OPS = [
+  ReducerOpRecord(lax.add, np.add, 0, default_dtypes, lax.reduce_sum_p),
+  ReducerOpRecord(lax.mul, np.multiply, 1, default_dtypes, lax.reduce_prod_p),
+  ReducerOpRecord(lax.max, np.maximum, 0, uint_dtypes + bool_dtypes, lax.reduce_max_p),
+  ReducerOpRecord(lax.max, np.maximum, -np.inf, float_dtypes, lax.reduce_max_p),
+  ReducerOpRecord(lax.max, np.maximum, dtypes.iinfo(np.int32).min, [np.int32], lax.reduce_max_p),
+  ReducerOpRecord(lax.max, np.maximum, dtypes.iinfo(np.int64).min, [np.int64], lax.reduce_max_p),
+  ReducerOpRecord(lax.min, np.minimum, np.inf, float_dtypes, lax.reduce_min_p),
+  ReducerOpRecord(lax.min, np.minimum, dtypes.iinfo(np.int32).max, [np.int32], lax.reduce_min_p),
+  ReducerOpRecord(lax.min, np.minimum, dtypes.iinfo(np.int64).max, [np.int64], lax.reduce_min_p),
+  ReducerOpRecord(lax.min, np.minimum, dtypes.iinfo(np.uint32).max, [np.uint32], lax.reduce_min_p),
+  ReducerOpRecord(lax.min, np.minimum, dtypes.iinfo(np.uint64).max, [np.uint64], lax.reduce_min_p),
+  ReducerOpRecord(lax.bitwise_and, np.bitwise_and, -1, int_dtypes + uint_dtypes + bool_dtypes, lax.reduce_and_p),
+  ReducerOpRecord(lax.bitwise_or, np.bitwise_or, 0, int_dtypes + uint_dtypes + bool_dtypes, lax.reduce_or_p),
+  ReducerOpRecord(lax.bitwise_xor, np.bitwise_xor, 0, int_dtypes + uint_dtypes + bool_dtypes, lax.reduce_xor_p),
+]
+
 
 class LaxTest(jtu.JaxTestCase):
   """Numerical tests for LAX operations."""
@@ -1715,28 +1736,28 @@ class LaxTest(jtu.JaxTestCase):
 
   @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_op={}_inshape={}_reducedims={}_initval={}"
-       .format(op.__name__, jtu.format_shape_dtype_string(shape, dtype), dims,
-               init_val),
-       "op": op, "init_val": init_val, "shape": shape, "dtype": dtype, "dims": dims}
-      for init_val, op, types in [
-          (0, lax.add, default_dtypes),
-          (1, lax.mul, default_dtypes),
-          (0, lax.max, all_dtypes), # non-monoidal
-          (-np.inf, lax.max, float_dtypes),
-          (dtypes.iinfo(np.int32).min, lax.max, [np.int32]),
-          (dtypes.iinfo(np.int64).min, lax.max, [np.int64]),
-          (np.inf, lax.min, float_dtypes),
-          (dtypes.iinfo(np.int32).max, lax.min, [np.int32]),
-          (dtypes.iinfo(np.int64).max, lax.min, [np.int64]),
-          (dtypes.iinfo(np.uint32).max, lax.min, [np.uint32]),
-          (dtypes.iinfo(np.uint64).max, lax.min, [np.uint64]),
-      ]
-      for dtype in types
+       .format(rec.op.__name__, jtu.format_shape_dtype_string(shape, dtype), dims,
+               rec.init_val),
+       "op": rec.op, "reference_op": rec.reference_op, "init_val": rec.init_val,
+       "shape": shape, "dtype": dtype, "dims": dims, "primitive": rec.primitive}
+      for rec in LAX_REDUCE_OPS
+      for dtype in rec.dtypes
       for shape, dims in [
           [(3, 4, 5), (0,)], [(3, 4, 5), (1, 2)],
           [(3, 4, 5), (0, 2)], [(3, 4, 5), (0, 1, 2)]
       ]))
-  def testReduce(self, op, init_val, shape, dtype, dims):
+  def testReduce(self, op, reference_op, init_val, shape, dtype, dims, primitive):
+    if not config.x64_enabled and dtype in (np.float64, np.int64, np.uint64):
+      raise SkipTest("x64 mode is disabled.")
+    def reference_fun(operand):
+      if hasattr(reference_op, "reduce"):
+        initial = np.array(init_val, dtype=dtype)
+        result = reference_op.reduce(operand, axis=dims, initial=initial)
+      else:
+        result = reference_op(operand, axis=dims)
+
+      return result.astype(dtype)
+
     rng_factory = (jtu.rand_default if dtypes.issubdtype(dtype, np.integer)
                    else jtu.rand_small)
     rng = rng_factory(self.rng())
@@ -1750,44 +1771,12 @@ class LaxTest(jtu.JaxTestCase):
     fun = lambda operand: lax.reduce(operand, init_val, op, dims)
     args_maker = lambda: [rng(shape, dtype)]
     self._CompileAndCheck(fun, args_maker)
-
-  @parameterized.named_parameters(jtu.cases_from_list(
-      {"testcase_name": "_op={}_shape={}_reducedims={}_initval={}_prim={}"
-       .format(op.__name__, shape, dims, init_val, prim),
-       "op": op, "init_val": init_val, "shape": shape, "dims": dims, "prim": prim}
-      for init_val, op , prim in [
-        (True, lax.bitwise_and, jax.lax.reduce_and_p),
-        (False, lax.bitwise_or, jax.lax.reduce_or_p),
-        (False, lax.bitwise_xor, jax.lax.reduce_xor_p),
-      ]
-      for shape, dims in [
-          [(3, 4, 5), (0,)], [(3, 4, 5), (1, 2)],
-          [(3, 4, 5), (0, 2)], [(3, 4, 5), (0, 1, 2)]
-      ]))
-  def testReduceBoolean(self, op, init_val, shape, dims, prim):
-    def reference_fun(operand, init_value):
-      np_op = getattr(np, op.__name__)
-      return np_op.reduce(operand, axis=dims, initial=init_val)
-
-    dtype = np.bool_
-    rng = jtu.rand_bool(self.rng())
-    init_val = np.asarray(init_val, dtype=dtype)
-    fun = lambda operand, init_val: lax.reduce(operand, init_val, op, dims)
-    args_maker = lambda: [rng(shape, dtype), init_val]
-    self._CompileAndCheck(fun, args_maker)
     self._CheckAgainstNumpy(reference_fun, fun, args_maker)
 
-    # recheck with a static init_val
-    fun = lambda operand: lax.reduce(operand, init_val, op, dims)
-    reference_fun = partial(reference_fun, init_value=init_val)
-    args_maker = lambda: [rng(shape, dtype)]
-    self._CompileAndCheck(fun, args_maker)
-    self._CheckAgainstNumpy(reference_fun, fun, args_maker)
-
-    # check that the correct monoid reducer primitive is used inside the
-    # jaxpr. This requires the init_val (monoid identity element) to be static
+    # check that the correct monoid reducer primitive is used inside the jaxpr.
+    # This requires the init_val (monoid identity element) to be static
     jaxpr = jax.make_jaxpr(fun)(rng(shape, dtype))
-    self.assertEqual(jaxpr.eqns[0].primitive, prim)
+    self.assertEqual(jaxpr.eqns[0].primitive, primitive)
 
   @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_op={}.{}_arr_weak_type={}_init_weak_type={}"

